### PR TITLE
Use Response constants for HTTP return codes

### PR DESCRIPTION
**Pull request by GawainLynch**
Dibs on early merge when we branch, before everyone starts pulling controllers apart.

### CODE REVIEW

File: src/Controllers/Async.php [hunk 1]
```diff
@@ -754,7 +754,7 @@ public function before(Request $request, Silex\Application $app)
 
         // If there's no active session, don't do anything.
         if (!$app['users']->isValidSession()) {
-            $app->abort(404, "You must be logged in to use this.");
+            $app->abort(Response::HTTP_NOT_FOUND, 'You must be logged in to use this.');
```
Comment (bobdenotter) [inline src/Controllers/Async.php]: Now that i look at this.. Is `404` the correct response code for this?

Comment (GawainLynch) [inline src/Controllers/Async.php]: One of the reasons I've changed these! It makes one look a bit closer :+1: 

But, yeah I'll update this one to HTTP_UNAUTHORIZED and yell if any others don't look right.

**Comment by CarsonF** [comment]
:+1:

**Comment by bobdenotter** [comment]
MERGEFEST!!

**Pull request merged**